src/main/java/hudson/tasks/junit/CaseResult.java

```diff
@@ -256,10 +262,25 @@ private static String getSkippedMessage(Element testCase) {
         return message;
     }
 
-    public String getDisplayName() {
+    public String getTransformedTestName() {
         return TestNameTransformer.getTransformedName(testName);
     }
 
+    public String getDisplayName() {
+        // Only prepend the enclosing flow node names if there are any and the run this is in has multiple blocks directly containing
+        // test results.
+        if (!getEnclosingFlowNodeNames().isEmpty()) {
```

Comment:
Well, it would mean `FlowNode.parents`, but it'd always be a max of 1. =) In any case, my current experimentation is only populating the `enclosingId` for `AtomNode` and `BlockStartNode`. I'll get a PR up tomorrow.